### DISCUSSION TITLE
Why is code from imported file not executed in server API route

### DISCUSSION THREAD

**Discussion by lewebsimple** [nuxt/nuxt#17731] [Questions]
I'm trying to define a modular GraphQL schema as suggested [here](https://github.com/hayes/pothos/discussions/273#discussioncomment-2046397).

Let's say we have server/api/hello.ts like so:
```
import '../test.ts`;
console.log('Hello from server/api/hello.ts');
export default () => 'Hello World';
```
...and server/test.ts only contains `console.log('Hello from server/test.ts');`

Why is the console only outputting "Hello from server/api/hello.ts" (twice?) and not executing the other console.log ?

Is tree shaking the cause of this behavior and if so, how can I bypass it ?

**Comment by Aareksio** [comment]
```js
// test.ts
export default function () {
  // your code here
}

// hello.ts
import test from '../test.ts'
test()
```

The imports are not executed top to bottom, only the code you actually call is executed.
